lib/Arch/AArch32_C.cpp

```diff
@@ -124,18 +124,7 @@ AArch32_C::BindReturnValues(llvm::Function &function, bool &injected_sret,
   if (function.hasStructRetAttr()) {
     auto &value_declaration = ret_values.emplace_back();
 
-    // Check both first and second parameter because llvm does that in
-    // llvm::Function::hasStructRetAttr()
-    if (function.hasParamAttribute(0, llvm::Attribute::StructRet)) {
-      value_declaration.type =
-          remill::NthArgument(&function, 0)->getType()->getPointerElementType();
-
-    } else if (function.hasParamAttribute(1, llvm::Attribute::StructRet)) {
-      value_declaration.type =
-          remill::NthArgument(&function, 1)->getType()->getPointerElementType();
-    }
-
-    value_declaration.type = llvm::PointerType::get(value_declaration.type, 0);
+    value_declaration.type = llvm::PointerType::get(function.getContext(), 0);
```

Comment:
I feel like I am a bit iffy on this change, but I can't quite say why. Can you try to formulate an example on compiler explorer that uses `structret` (e.g. have a function return a big `struct` by value), and then show the llvm 12/13 and trunk llvm bitcode side-by-side, and then also make two json specs, one using the old build, one with the new build, and then see what bitcode we produce and how it relates to the original?

Reply:
Can be for x86-64, doesn't need to be aarch32.

Reply:
I see you marked this as resolved. Do you still want me to do this?